packages/react-native/React/CoreModules/RCTAppearance.mm

```diff
@@ -39,10 +39,15 @@ void RCTOverrideAppearancePreference(NSString *const colorSchemeOverride)
 static UITraitCollection *getKeyWindowTraitCollection()
 {
   __block UITraitCollection *traitCollection = nil;
-  RCTExecuteOnMainQueue(^{
-    traitCollection = RCTSharedApplication().delegate.window.traitCollection;
-  });
-  return traitCollection;
+  if (RCTIsMainQueue()) {
+    return RCTSharedApplication().delegate.window.traitCollection;
+  } else {
+    __block UITraitCollection* traitCollection = nil;
+    dispatch_sync(dispatch_get_main_queue(), ^{
```

Comment:
We should avoid using dispatch_sync as much as possible, as it can lead to deadlocks.

This module is marked as `requiresMainQueueSetup`, which means the constructor is called from the main thread. Is there any way we can access this attribute there?

Also, as applies to most classes in UIKit, this class is not safe to use outside the main thread. We should grab just the `userInterfaceStyle` prop, and not leak anything else.

Reply:
Something like this? 
```objc
- (instancetype)init
{
  if ((self = [super init])) {
    UITraitCollection *traitCollection = RCTSharedApplication().delegate.window.traitCollection;
    _currentColorScheme = RCTColorSchemePreference(traitCollection);
  }
  return self;
}
```